### PR TITLE
Feature/#812 update kafka faas connector deployment info

### LINKED ISSUE #812: Add endpoint for updating the KafkaFaasConnector deployment information
# Description
Adds the following endpoint
- `PUT /applications/<short-name>/<version>/kafkaFaasConnectorDeploymentInformation/`

- [ ] this PR contains breaking changes!

# Resolves / is related to

Relates to #750
Relates to #798

# What is affected by this PR

- [x] Backend
    - [ ] Kubernetes logic
    - [ ] Domain model
- [x] API
    - [x] discussed changes in the team / informed the team about changes
    - [ ] updated [`insertTestValues.sh`](https://github.com/UST-MICO/mico/blob/master/insertTestValues.sh) and [Postman Collections](https://github.com/UST-MICO/docs/tree/master/debugging-testing/postman) are updated
- [ ] Frontend
    - [ ] Ensure that it is compilable for production (`npm run build -- --prod`)
- [ ] Documentation

# Checklist

- [x] Ensure that new source files include the [license header](https://github.com/UST-MICO/mico/blob/master/CONTRIBUTING.md#source-file-headers)
- [ ] Ensure [JavaDoc is up to date](https://github.com/UST-MICO/mico/tree/master/docs#build-the-documentation-locally)

### PR DESCRIPTION
**Pull request by pascalweiss**
# Description
Adds a new PUT request to the following endpoint: ```http://<host>/applications/<shortName>/<version>/kafkaFaasConnectorDeploymentInformation/<instanceId>```
The following properties of a kfConnectorDeploymentInfo can be updated:
- input topic name
- output topic name
- OpenFaaS function name

<br/>
The OpenFaaS Function name is stored, like MicoTopics, as a separate node. 

- [ ] this PR contains breaking changes!

# Resolves / is related to
Relates to #750, #798
Closes #812 

# What is affected by this PR

- [x] Backend
    - [ ] Kubernetes logic
    - [x] Domain model
- [x] API
    - [ ] discussed changes in the team / informed the team about changes
    - [ ] updated [`insertTestValues.sh`](https://github.com/UST-MICO/mico/blob/master/insertTestValues.sh) and [Postman Collections](https://github.com/UST-MICO/docs/tree/master/debugging-testing/postman) are updated
- [ ] Frontend
    - [ ] Ensure that it is compilable for production (`npm run build -- --prod`)
- [ ] Documentation

# Checklist

- [ ] Ensure that new source files include the [license header](https://github.com/UST-MICO/mico/blob/master/CONTRIBUTING.md#source-file-headers)
- [ ] Ensure [JavaDoc is up to date](https://github.com/UST-MICO/mico/tree/master/docs#build-the-documentation-locally)

### CODE REVIEW

File: mico-core/src/main/java/io/github/ust/mico/core/broker/KafkaFaasConnectorDeploymentInfoBroker.java [hunk 1]
```diff
@@ -83,4 +93,86 @@
...
+        for (MicoTopicRole role : topics) {
+            MicoTopic topic = role.getTopic();
+            switch (role.getRole()) {
+                case INPUT:
+                    topic.setName(kfConnectorDeploymentInfoRequestDTO.getInputTopicName());
+                    break;
+                case OUTPUT:
+                    topic.setName(kfConnectorDeploymentInfoRequestDTO.getOutputTopicName());
+                    break;
+                case DEAD_LETTER: //TODO
```
Comment (davidkopp) [inline mico-core/src/main/java/io/github/ust/mico/core/broker/KafkaFaasConnectorDeploymentInfoBroker.java]: Only `INPUT` and `OUTPUT` are relevant

File: mico-core/src/main/java/io/github/ust/mico/core/broker/KafkaFaasConnectorDeploymentInfoBroker.java [hunk 2]
```diff
@@ -83,4 +93,86 @@
...
+    private MicoServiceDeploymentInfo saveValuesToDatabase(KFConnectorDeploymentInfoRequestDTO kfConnectorDeploymentInfoRequestDTO,
+                                                           MicoServiceDeploymentInfo storedServiceDeploymentInfo) {
+
+        eventuallyUpdateTopics(kfConnectorDeploymentInfoRequestDTO, storedServiceDeploymentInfo);
+        eventuallyUpdateOpenFaaSFunction(kfConnectorDeploymentInfoRequestDTO, storedServiceDeploymentInfo);
+        deploymentInfoRepository.save(storedServiceDeploymentInfo);
+        return storedServiceDeploymentInfo;
+    }
+
+    private void eventuallyUpdateTopics(KFConnectorDeploymentInfoRequestDTO kfConnectorDeploymentInfoRequestDTO,
```
Comment (buehlefs) [inline mico-core/src/main/java/io/github/ust/mico/core/broker/KafkaFaasConnectorDeploymentInfoBroker.java]: This function is complex enought to warrant javadoc!

File: mico-core/src/main/java/io/github/ust/mico/core/broker/KafkaFaasConnectorDeploymentInfoBroker.java [hunk 3]
```diff
@@ -83,4 +93,86 @@
...
+                        .setServiceDeploymentInfo(storedServiceDeploymentInfo)
+                        .setTopic(new MicoTopic().setName(
+                            (role == INPUT) ? kfConnectorDeploymentInfoRequestDTO.getInputTopicName()
+                                : kfConnectorDeploymentInfoRequestDTO.getOutputTopicName()
+                        )));
+            }
+        }
+    }
+
+    private void eventuallyUpdateOpenFaaSFunction(KFConnectorDeploymentInfoRequestDTO kfConnectorDeploymentInfoRequestDTO,
```
Comment (buehlefs) [inline mico-core/src/main/java/io/github/ust/mico/core/broker/KafkaFaasConnectorDeploymentInfoBroker.java]: This should also get javadoc.

File: mico-core/src/main/java/io/github/ust/mico/core/broker/KafkaFaasConnectorDeploymentInfoBroker.java [hunk 4]
```diff
@@ -83,4 +93,93 @@
...
+                                                  MicoServiceDeploymentInfo storedServiceDeploymentInfo) {
+        OpenFaaSFunction openFaaSFunction = storedServiceDeploymentInfo.getOpenFaaSFunction();
+        if (openFaaSFunction == null) {
+            storedServiceDeploymentInfo.setOpenFaaSFunction(
+                new OpenFaaSFunction().setName(kfConnectorDeploymentInfoRequestDTO.getOpenFaaSFunctionName()));
```
Comment (davidkopp) [inline mico-core/src/main/java/io/github/ust/mico/core/broker/KafkaFaasConnectorDeploymentInfoBroker.java]: If a node with this OpenFaaSFunction already exists in the database, it should be reused.

File: mico-core/src/main/java/io/github/ust/mico/core/broker/MicoServiceDeploymentInfoBroker.java [hunk 5]
```diff
@@ -272,6 +277,18 @@ private MicoServiceDeploymentInfo createOrReuseTopics(MicoServiceDeploymentInfo
         return serviceDeploymentInfo;
     }
 
+    MicoServiceDeploymentInfo createOrReuseOpenFaaSFunctionsInDatabase(MicoServiceDeploymentInfo serviceDeploymentInfo) {
```
Comment (buehlefs) [inline mico-core/src/main/java/io/github/ust/mico/core/broker/MicoServiceDeploymentInfoBroker.java]: Missing Javadoc

**Pull request merged**